ironfish/src/consensus/consensus.ts

```diff
@@ -61,18 +62,29 @@ export type ConsensusParameters = {
    * allowing for a greater per-block downward shift.
    */
   enableIncreasedDifficultyChange: ActivationSequence
+
+  /**
+   * Mapping of block height to the hash of the block at that height. Once a node has added this block to
+   * its main chain, it will not be disconnected from the main chain.
+   */
+  checkpoints: Checkpoint[]
 }
 
 export class Consensus {
   readonly parameters: ConsensusParameters
+  readonly checkpoints: Map<number, Buffer>
 
   constructor(parameters: ConsensusParameters) {
     this.parameters = parameters
+    this.checkpoints = new Map<number, Buffer>()
```

Comment:
Can you add some tests here around the new logic in the constructor and also the isActive function? Doesn't need to be too crazy, but its finnicky enough that it is liable to silently break in the future